package.json

```diff
@@ -8,13 +8,15 @@
     "lint": "next lint"
   },
   "dependencies": {
-    "next": "12.0.3",
+    "next": "^12.0.4-canary.11",
```

Comment:
Based on the following GH Discussion: https://github.com/vercel/next.js/discussions/30174#discussion-3643870

> Styled Components (update: Nov 10 2021)
The most important part of babel-plugin-styled-components for Next.js support has been ported to SWC, namely the ssr and displayName options, both are enabled by default in the Babel plugin. The ssr option ensures there is no hydration mismatch between client/server

> This is out now as experimental on next@canary (npm install next@canary). The flags have good defaults for Next.js so styledComponents is just an on/off boolean:

> ```js
> // next.config.js
> module.exports = {
>  experimental: {
>    // Enables the styled-components SWC transform
>    styledComponents: true
>  }
> }
> ```

Reply:
Alternatively, a combination of  [babel-plugin-styled-components](https://github.com/vercel/next.js/blob/e969d226999bb0fcb52ecc203b359f3715ff69bf/examples/with-typescript-styled-components/package.json#L21) and a [.babelrc](https://github.com/vercel/next.js/blob/e969d226999bb0fcb52ecc203b359f3715ff69bf/examples/with-typescript-styled-components/.babelrc#L1-L4) could be used, but Next.js will automatically opt-out of using the SWC compiler.